icu4c/source/common/unicode/unistr.h

```diff
@@ -3600,6 +3600,10 @@ class U_COMMON_API UnicodeString : public Replaceable
   static inline UnicodeString readOnlyAlias(const S &text) {
     return readOnlyAliasFromU16StringView(internal::toU16StringView(text));
   }
+  /** @internal */
```

Comment:
It doesn't seem right to declare this internal, when we expect this version to be called for UnicodeString inputs.
Please copy the API docs from the previous new function (with adjustments) and send an addendum email reply to the icu-design thread.

Reply:
I thought of disambiguating overloads as an implementation detail, but I've changed the docstring now.